src/texture/mod.rs

```diff
@@ -358,6 +358,38 @@ pub struct RawImage3d<'a, T: Clone + 'a> {
     pub format: ClientFormat,
 }
 
+impl<'a, T: Clone + 'a> RawImage3d<'a, T> {
+    ///Transforms a Vec<RawImage2d> into a RawImage3d
+    pub fn from_vec_raw2d(arr: &Vec<RawImage2d<'a, T>>) -> Result<RawImage3d<'a, T>, TextureMaybeSupportedCreationError> {
+        let depth   = arr.len();
+        let width   = arr[0].width;
+        let height  = arr[0].height;
+        let format  = arr[0].format;
+        let raw_data = {
+            let mut vec = Vec::<T>::with_capacity((width * height * depth as u32) as usize);
+            for i in arr {
+                if width != i.width || height != i.height {
+                    return Err(TextureMaybeSupportedCreationError::CreationError(TextureCreationError::DimensionsNotSupported));
```

Comment:
The error here describes dimensions and formats that are not supported by the hardware, so that's irrevelant here.

The user is supposed to pass images that have identical dimensions. If it's not the case, it's clearly the user's fault.